field_map.go

```diff
@@ -49,12 +51,16 @@ func (m *FieldMap) init() {
 }
 
 func (m *FieldMap) initWithOrdering(ordering tagOrder) {
+	m.rwLock = &sync.RWMutex{}
```

Comment:
I guess one concern is that we need to make sure structs of type `FieldMap` are initialized properly by calling this. 
Otherwise we might end up with some panics. 

For example currently there are structs such as 
```go
type Header struct{ FieldMap }
```
but from what I can see the existing structs are initialized properly by having something like the below and then calling it.
```go
func (h *Header) Init() {
	h.initWithOrdering(headerFieldOrdering)
}
```

So for the existing structs looks like we should not have a problem, but new ones can still be problematic.   
Don't have a perfect solutions for this though, so we can keep it as it's for now.

Reply:
nvm, missed your comment in PR description that we are simply copying the commit from original repo.